packages/wallets/src/features/cashier/modules/DepositFiat/__tests__/DepositFiat.spec.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { useAuthorize, useCashierFiatAddress } from '@deriv/api';
+import { act, render, screen, waitFor } from '@testing-library/react';
+import DepositFiat from '../DepositFiat';
+
+jest.mock('@deriv/api', () => ({
+    useAuthorize: jest.fn(),
+    useCashierFiatAddress: jest.fn(),
+}));
+
+describe('DepositFiat', () => {
+    beforeEach(() => {
+        (useAuthorize as jest.Mock).mockReturnValueOnce({ isSuccess: true });
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('should call useAuthorize and useCashierFiatAddress hooks on mount', () => {
```

Comment:
question about the general patterns: should we test what component does (I mean rendering, so in a way, testing the input/output of it), or should we also test how component works internally? 

IMO default should be to test that component renders whats needed. 
If I change hook from useAuthorize to something else, or change useCashierFiatAdress, ideally, this test should still work,

Reply:
if we gonna be testing internals of component logic, they gonna be way harder to refactor, than if we gonna be testing just the input (props) and output (rendered stuff). 

So as a rule of thumb, we should test input/output of component. Internals of component should be tested only when its actually needed - for example: "component unsubscribes from API to avoid memory leaks" (although even that case - with proper hooks / proper API, should be needed)

Reply:
Hi @wojciech-deriv, you're right ideally we should be testing the input/output of the components and changing the internal logic shouldn't be breaking the tests. I'll remove the first test case as it is unneeded 👍